src/extension.ts

```diff
@@ -207,6 +224,62 @@ async function openPackageDirectoryCommand() {
     }
 }
 
+async function weaveCommand() {
+    if (vscode.window.activeTextEditor === undefined) {
+        vscode.window.showErrorMessage('Please open a document before you execute the weave command.');
+    }
+    else if (vscode.window.activeTextEditor.document.isDirty || vscode.window.activeTextEditor.document.isUntitled) {
+        vscode.window.showErrorMessage('Please safe the current file before you weave it.');
+    }
+    else if (vscode.window.activeTextEditor.document.languageId!='julia' && vscode.window.activeTextEditor.document.languageId!='juliamarkdown') {
+        vscode.window.showErrorMessage('Only julia (.jl) files can be weaved.')
+    }
+    else {
+        let parsed_filename = path.parse(vscode.window.activeTextEditor.document.fileName);
+
+        let output_filename = path.join(parsed_filename.dir, parsed_filename.name + ".html");
+
+        if (weaveOutputChannel == null) {
+            weaveOutputChannel = vscode.window.createOutputChannel("julia weave");
+        }
+        weaveOutputChannel.clear();
+        weaveOutputChannel.show(true);
+
+        if (weaveChildProcess != null) {
+            try {
+                await kill(weaveChildProcess);
+            }
+            catch (e) {
+            }
+        }
+
+        weaveChildProcess = spawn(juliaExecutable, ['-e', 'using Weave; Weave.weave(Base.ARGS[1], out_path=:doc)', vscode.window.activeTextEditor.document.fileName]);
+        weaveChildProcess.stdout.on('data', function (data) {
+            weaveOutputChannel.append(String(data));
+        });
+        weaveChildProcess.stderr.on('data', function (data) {
+            weaveOutputChannel.append(String(data));
+        });
+        weaveChildProcess.on('close', async function (code) {
+            weaveChildProcess = null;
+
+            if (code==0) {
+                weaveOutputChannel.hide();
+
+                lastWeaveContent = await fs.readFile(output_filename, "utf-8")
+
+                let uri = vscode.Uri.parse('jlweave://nothing.html');
+                let success = await vscode.commands.executeCommand('vscode.previewHtml', uri, vscode.ViewColumn.One, "julia weave output");
```

Comment:
I thought about it, but I think it would set up an expectation about a live update of the preview, which we don't have... We could make it an option down the road.